jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java

```diff
@@ -151,13 +181,12 @@ protected Connection acquire(boolean create)
      * @param maxPending the max desired number of connections scheduled for opening,
      * or a negative number to always trigger the opening of a new connection
      */
-    protected void tryCreate(int maxPending)
+    void tryCreate(int maxPending)
     {
         while (true)
         {
-            long encoded = connections.get();
-            int pending = AtomicBiInteger.getHi(encoded);
-            int total = AtomicBiInteger.getLo(encoded);
+            int pending = pendingConnectionCount.get();
+            int total = pool.size() + pending;
```

Comment:
`AtomicBiInteger` is my current favourite class... at least until we add `AtomicQuadShort`, which will then be my favourite class until we add `AtomicOctaByte`!      So don't think you can get rid of it that easily.

The point of encoding both the size and pending into the same Atomic is so that one cannot be changed while we are looping/considering/changing the other.    In this case, we might decide to increment pending but concurrently the size is already incremented to the max size.  Previously we would see that in 171 and loop, now we don't see it.

I so have to find a need for `AtomicOctaByte`.... maybe that would simplify `HttpInput`?

Reply:
This still needs to be resolved.